app/scripts/modules/core/src/notification/modal/EditNotificationModal.tsx

```diff
@@ -12,13 +12,29 @@ export interface IEditNotificationModalProps extends IModalComponentProps {
   level: string;
   notification: INotification;
   stageType: string;
+  editNotification: (n: INotification) => Promise<INotification[]>;
 }
 
-export class EditNotificationModal extends React.Component<IEditNotificationModalProps> {
+export interface IEditNotificationModalState {
+  isSubmitting: boolean;
+}
+
+export class EditNotificationModal extends React.Component<IEditNotificationModalProps, IEditNotificationModalState> {
+  constructor(props: IEditNotificationModalProps) {
+    super(props);
+    this.state = {
+      isSubmitting: false,
+    };
+  }
+
   private formikRef = React.createRef<Formik<any>>();
 
   private submit = (values: INotification): void => {
-    this.props.closeModal(values);
+    this.setState({ isSubmitting: true });
+    this.props.editNotification(values).then(() => {
+      this.setState({ isSubmitting: false });
+      this.props.closeModal();
```

Comment:
fyi @christopherthielen, the modal should now close after the edits are made. The angular component was saving notifications async and the updates would eventually propagate. Now, `editNotification` and `saveNotifications` return promises so we can chain `closeModal` to the end.